nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RetryCount.java

```diff
@@ -0,0 +1,210 @@
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.apache.nifi.processors.standard.RetryCount.RETRY_COUNT_ATTRIBUTE_KEY;
+import static org.apache.nifi.annotation.behavior.InputRequirement.Requirement.INPUT_REQUIRED;
+import static org.apache.nifi.expression.ExpressionLanguageScope.VARIABLE_REGISTRY;
+
+/**
+ * Created by jpercivall on 3/17/17.
+ */
+@SupportsBatching
+@SideEffectFree
+@InputRequirement(INPUT_REQUIRED)
+@Tags({"penalize", "retry", "penalize"})
+@CapabilityDescription("Used to facilitate retrying a FlowFile a set number of times before considering the FlowFile 'failed'. Can also be used as a utility to penalize FlowFiles.")
+@WritesAttribute(attribute = RETRY_COUNT_ATTRIBUTE_KEY, description = "The number of times this FlowFile has been routed to 'Retry'.")
+@ReadsAttribute(attribute = RETRY_COUNT_ATTRIBUTE_KEY, description = "The number of times this FlowFile has been routed to 'Retry'.")
+public class RetryCount extends AbstractProcessor {
+
+    public static final String RETRY_COUNT_ATTRIBUTE_KEY = "retryCountProcessor.timesRetried";
+
+    public static final PropertyDescriptor PROP_RETRY_LIMIT = new PropertyDescriptor.Builder()
+            .name("Retry limit")
+            .displayName("Retry limit")
+            .description("The number of times to retry the FlowFile before considering it 'failed' and routing it to 'over limit'.")
+            .expressionLanguageSupported(VARIABLE_REGISTRY)
+            .required(true)
+            .addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .defaultValue("3")
+            .build();
+
+    public static final PropertyDescriptor PROP_PENALIZE_FLOWFILE = new PropertyDescriptor.Builder()
+            .name("Penalize FlowFile")
+            .displayName("Penalize FlowFile")
+            .description("If true then the FlowFiles routed to 'retry' will be penalized.")
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .build();
+
+    public static final PropertyDescriptor PROP_WARN_ON_OVER_LIMIT = new PropertyDescriptor.Builder()
+            .name("Warn on 'over limit'")
+            .displayName("Warn on 'over limit'")
+            .description("If true then when a FlowFile is routed to 'over limit' a message will be logged at the level 'warn'.")
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .build();
+
+    public static final PropertyDescriptor PROP_INPUT_QUEUE_DATA_SIZE_LIMIT = new PropertyDescriptor.Builder()
+            .name("Input queue data size limit")
```

Comment:
Suggestion: `retry-input-queue-size-limit`
This is a cool feature, looking forward to testing it.